File: src/service/ModelBuilder.Models/Application/ConnectorAm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using Mb.Models.Enums;
using Mimirorg.TypeLibrary.Enums;

namespace Mb.Models.Application;

//#region ConnectorAm

public class ConnectorAm
{
    [Required]
    public Guid Id { get; set; }
    [Required]
    public string Name { get; set; }    
    [Required]
    public ConnectorDirection Direction { get; set; }
    [Required]
    public Guid Inside { get; set; }
    [Required]
    public Guid Outside { get; set; }
    [Required]
    public Guid Block { get; set; }
    [Required]
    public RelationType Discriminator { get; set; } = RelationType.NotSet;

    //public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
    //{
    //    return new List<ValidationResult>();
    //}
}

//#endregion ConnectorAm

//#region ConnectorTerminalAm

//public class ConnectorTerminalAm : ConnectorAm, IValidatableObject
//{
//    //[Required]
//    //public string TerminalType { get; set; }    
//    //public string ReferenceType { get; set; }
//    //public ICollection<AttributeAm> Attributes { get; set; }

//    //public new IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
//    //{
//    //    var validations = new List<ValidationResult>();

//    //    if (Attributes != null)
//    //        validations.AddRange(Attributes.SelectMany(attributeAm => attributeAm.Validate(validationContext)));

//    //    return validations;
//    //}
//}

//#endregion ConnectorTerminalAm

//#region ConnectorRelationAm

//public abstract class ConnectorRelationAm : ConnectorAm
//{
//}

//#endregion ConnectorRelationAm

//#region ConnectorFulfilledByAm

//public class ConnectorFulfilledByAm : ConnectorRelationAm
//{
//}

//#endregion ConnectorFulfilledByAm

//#region ConnectorHasLocationAm

//public class ConnectorHasLocationAm : ConnectorRelationAm
//{
//}

//#endregion ConnectorHasLocationAm

//#region ConnectorPartOfAm

//public class ConnectorPartOfAm : ConnectorRelationAm
//{
//}

//#endregion ConnectorPartOfAm

File: src/client/src/lib/classes/application/ConnectorAm.ts
import { Direction } from "../../enums";
import {
  Connector,
  ConnectorFulfilledBy,
  ConnectorHasLocation,
  ConnectorPartOf,
  ConnectorRelation,
  ConnectorTerminal,
} from "../Connector";
import { AttributeAm } from "./AttributeAm";

export class ConnectorAm {
  id: string;
  name: string;
  inside: string;
  outside: string;
  direction: Direction;
  aspectObject: string;

  public constructor(obj: Connector) {
    this.id = obj.id;
    this.name = obj.name;
    this.inside = obj.inside;
    this.direction = obj.direction;
    this.aspectObject = obj.aspectObject;
  }
}

export class ConnectorTerminalAm extends ConnectorAm {
  color: string;
  attributes: AttributeAm[];
  referenceType: string;

  public constructor(obj: ConnectorTerminal) {
    super(obj);
    this.color = obj.color;
    this.attributes = obj.attributes.map((attribute) => new AttributeAm(attribute));
    this.referenceType = obj.referenceType;
  }
}

export class ConnectorRelationAm extends ConnectorAm {
  public constructor(obj: ConnectorRelation) {
    super(obj);
  }
}

export class ConnectorPartOfAm extends ConnectorRelationAm {
  public constructor(obj: ConnectorPartOf) {
    super(obj);
  }
}

export class ConnectorFulfilledByAm extends ConnectorRelationAm {
  public constructor(obj: ConnectorFulfilledBy) {
    super(obj);
  }
}

export class ConnectorHasLocationAm extends ConnectorRelationAm {
  public constructor(obj: ConnectorHasLocation) {
    super(obj);
  }
}
